File: quiz/boj/11055/main.py
n =int(input())
arr = [int(v) for v in input().split()]

# D[i] = i원소를 끝으로 하는 수열의 최대 합

# 자기 자신으로 초기화
# 조심!! D = arr는 값 복사가 아니라 래퍼런스를 덮어씌워버림. 
# 따라서, D[i]를 변경하면 arr[i]도 변한다!!!!
# 그러므로 arr[:]를 써야한다

D = arr[:]

for i in range(1, n):
    for j in range(i):
        if arr[j] < arr[i]:
            D[i] = max(D[j]+arr[i], D[i])
    

print(max(D))

File: quiz/boj/11055/main.js
const readline = require("readline");
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

T = 0;
isFirst = true;
arr = [];

rl.on("line", function(line) {
    if(isFirst){
        T = line;
        isFirst = false;
        return;
    }
    arr = line.split(' ').map(function(num){ return parseInt(num, 10)});
    rl.close();
}).on("close", function() {
    
    solution(arr, T);
  process.exit();
});

function solution(arr, size){
    var dp = arr.slice();

    for(var i=1; i<size; i++){
        for(var j=0; j<i; j++){
            if(arr[j]< arr[i]){
                dp[i] = Math.max(dp[i], dp[j]+arr[i]);
            }
            
        }
    }
    
    
    
    // var result = Math.max(...dp);
    var result= dp.reduce(function(prev, cur){ return prev> cur ? prev: cur; });
    
    
    console.log(result);
    
}
